src/components/ec-btn/ec-btn.vue

```diff
@@ -50,150 +50,115 @@
+  isRounded: false,
+  isOutline: false,
+  isFullWidth: false,
+  isReverse: false,
+  isLoading: false,
+  isSubmit: true,
 });
 
-const {
-  size,
-  href,
-  to,
-  tag,
-  isDisabled,
-  icon,
-  isRounded,
-  isOutline,
-  isFullWidth,
-  category,
-  isReverse,
-  isLoading,
-  isSubmit,
-} = toRefs(props);
-
-const isRouterLink = computed(() => !!to.value);
-const isAnchorLink = computed(() => !!href.value);
+const isRouterLink = computed(() => !!props.to);
+const isAnchorLink = computed(() => !!props.href);
 
 const componentTag = computed(() => {
   if (isAnchorLink.value) {
-    return tag.value || 'a';
+    return props.tag || 'a';
   } if (isRouterLink.value) {
-    return tag.value || 'router-link';
+    return props.tag || 'router-link';
   }
-  return tag.value || 'button';
+  return props.tag || 'button';
 });
 
 const componentProps = computed(() => {
-  const newProps = {};
-  if (!isAnchorLink.value && (isDisabled.value || isLoading.value)) {
-    newProps.disabled = true;
-  }
-  if (componentTag.value === 'button') {
-    newProps.type = isSubmit.value ? 'submit' : 'button';
-  }
-  if (to.value) {
-    newProps.to = to.value;
-  }
-  if (href.value) {
-    newProps.href = href.value;
+  const isDisabled = props.isDisabled || props.isLoading;
+
+  switch (componentTag.value) {
+    case 'button':
+      return {
+        disabled: isDisabled || undefined,
+        type: props.isSubmit ? 'submit' : 'button',
+      } as HTMLButtonElement;
+    case 'router-link':
+      return {
+        disabled: isDisabled || undefined,
+        to: props.to,
+      } as RouterLinkProps;
+    case 'a': return {
+      href: props.href,
+    } as HTMLLinkElement;
+    default: return {
+      href: props.href,
+      to: props.to,
+    } as unknown as HTMLElement;
   }
-  return newProps;
 });
 
 function isSpinnerLoaderVisible() {
-  return isLoading.value && !slots['loading-text'];
+  return props.isLoading && !slots['loading-text'];
 }
 function isTextLoaderVisible() {
-  return isLoading.value && !!slots['loading-text'];
+  return props.isLoading && !!slots['loading-text'];
 }
 function hasDefaultSlot() {
   return !!slots.default;
 }
 
 function getButtonClasses() {
   return {
-    'ec-btn--sm': size.value === 'sm',
-    'ec-btn--md': size.value === 'md',
-    'ec-btn--rounded': isRounded.value,
-    'ec-btn--full-width': isFullWidth.value,
-    'ec-btn--icon-only': icon.value && !hasDefaultSlot(),
-    [`ec-btn--${category.value}`]: category.value,
-    'ec-btn--outline': isOutline.value,
-    'ec-btn--primary-reverse': isReverse.value && category.value === 'primary',
-    'ec-btn--secondary-reverse': isReverse.value && category.value === 'secondary',
-    'ec-btn--success-reverse': isReverse.value && category.value === 'success',
-    'ec-btn--error-reverse': isReverse.value && category.value === 'error',
-    'ec-btn--warning-reverse': isReverse.value && category.value === 'warning',
+    'ec-btn--sm': props.size === 'sm',
```

Comment:
Shall we use the enums here? `ButtonSize.Sm`, and few lines below `ButtonCategory.Primary` etc?

Reply:
Thank you! Sure we must!